src/pypi_attestation_models/_cli.py

```diff
@@ -0,0 +1,302 @@
+from __future__ import annotations
+
+import argparse
+import json
+import logging
+from pathlib import Path
+from typing import NoReturn
+
+from cryptography import x509
+from pydantic import ValidationError
+from sigstore.oidc import IdentityError, IdentityToken, Issuer, detect_credential
+from sigstore.sign import SigningContext
+from sigstore.verify import Verifier, policy
+
+from pypi_attestation_models import __version__
+from pypi_attestation_models._impl import Attestation, VerificationError
+
+logging.basicConfig(format="%(message)s", datefmt="[%X]", handlers=[logging.StreamHandler()])
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.INFO)
+
+
+def _parser() -> argparse.ArgumentParser:
+    parent_parser = argparse.ArgumentParser(add_help=False)
+
+    parent_parser.add_argument(
+        "-v",
+        "--verbose",
+        action="count",
+        default=0,
+        help="run with additional debug logging; supply multiple times to increase verbosity",
+    )
+
+    parser = argparse.ArgumentParser(
+        prog="pypi-attestation-models",
+        description="Sign, inspect or verify PEP 740 attestations generated for Python Packages",
+        parents=[parent_parser],
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-V",
+        "--version",
+        action="version",
+        version=f"pypi-attestation-models {__version__}",
+    )
+
+    subcommands = parser.add_subparsers(
+        required=True,
+        dest="subcommand",
+        metavar="COMMAND",
+        help="The operation to perform",
+    )
+
+    sign_command = subcommands.add_parser(
+        name="sign", help="Sign one or more inputs", parents=[parent_parser]
+    )
+
+    sign_command.add_argument(
+        "--identity-token",
+        type=str,
+        help="Identity token to use",
+        required=False,
+    )
```

Comment:
CC @facutuesca for opinions as well, but I'm inclined to think we can remove this -- for this CLI, the only ways to obtain an OIDC credential should be (1) through ambient detection, or (2) an interactive OAuth flow, with the logic being to always try (1) and then fallback to (2). 

(My rationale: I think in practice users won't need this, since 99.9% of them should go through ambient credentials instead, and the 0.1% is mostly for local non-CI testing.)

Reply:
I think it's fine to remove it, but we will have to take the audience as a parameter instead, and depend on `id` to detect the ambient credential. 
The tradeoff is flexibility (taking the token means we can run this outside CI/CD) vs ease of use (taking the audience is more user friendly, since the user does not need to care about `id`).

I'm OK with either, so removing it is fine.

Reply:
As the token as a short validation period, it is annoying to generate and use before it gets expired. It was useful for the tests though ( [like here](https://github.com/trailofbits/pypi-attestation-models/pull/22/files#diff-ff96f3e5f9d5e6247ecc149b49040031efb6137fed2399a201553abb541e57a7R78-R79) )

Reply:
>  It was useful for the tests though

We can still run those tests, we just add a pytest marker so that they only run in CI (similar to how `@online` works)

Reply:
Parameter deleted.